Objects/typeobject.c

```diff
@@ -7858,6 +7899,9 @@ type_ready_fill_dict(PyTypeObject *type)
     if (type_dict_set_doc(type) < 0) {
         return -1;
     }
+    if (cleanup_tp_dict(type) < 0) {
+        return -1;
+    }
```

Comment:
Let's move the top-level logic (and explanation) here, to simplify the new helper function.

```suggestion
    /* For now, we work around issues with slot wrappers
       for builtin types in subinterpreters.
       See https://github.com/python/cpython/issues/117482. */
    if (type->tp_flags & _Py_TPFLAGS_STATIC_BUILTIN) {
        PyInterpreterState *interp = _PyInterpreterState_GET();
        if (!_Py_IsMainInterpreter(interp)) {
            if (fix_builtin_slot_wrappers(type, interp) < 0) {
                return -1;
            }
        }
    }
```